client/src/components/ViewPerk.svelte

```diff
@@ -0,0 +1,53 @@
+<script>
```

Comment:
I dont think its necessary to have a whole new view and controller for the Card view. Instead if you show most of the information on the card in the perk manager, you can just click it to show the back of the card, which will show the code. That way you can combine into one view and minimize the risk of the faulty api routes